src/common.ts

```diff
@@ -15,8 +15,13 @@ process.on("uncaughtException", (e) => {
 });
 
 const cwd = core.getInput("working-directory");
+
+// Read each line of target-dir as a unique target directory
 // TODO: this could be read from .cargo config file directly
-const targetDir = core.getInput("target-dir") || "./target";
+const targetDirInput = core.getInput("target-dir") || "./target";
+const targetDirs = targetDirInput.trim().split("\n");
+core.info("Using target dirs: " + JSON.stringify(targetDirs));
```

Comment:
we already print an info of all the directories that are being cached, that also implies these.